cdap-watchdog/src/main/java/io/cdap/cdap/logging/ErrorLogsClassifier.java

```diff
@@ -91,10 +101,20 @@ private static boolean isFailureDetailsProviderInstance(String className) {
    * Constructor for {@link ErrorLogsClassifier}.
    */
   @Inject
-  public ErrorLogsClassifier(MetricsCollectionService metricsCollectionService) {
+  public ErrorLogsClassifier(CConfiguration cConf,
+      MetricsCollectionService metricsCollectionService) {
     this.metricsCollectionService = metricsCollectionService;
     responseCache = CacheBuilder.newBuilder().maximumSize(1000)
         .expireAfterWrite(7, TimeUnit.DAYS).build();
+    String rulesFilePath = cConf.get(RULES_FILE_PATH_KEY);
+    Type listType = new TypeToken<List<ErrorClassificationRule>>() {}.getType();
+    try (Reader reader = new FileReader(rulesFilePath)) {
+      ruleList = GSON.fromJson(reader, listType);
+    } catch (Exception e) {
+      LOG.info("Unable to read rules list file.", e);
```

Comment:
A mis configuration should be a fatal error. It's better not to start at all than ignore a configuration. You can skip it if it's not set

Reply:
I don't want logging service to go down for this issue, which is why I thought not to fail it.

Reply:
Well, it's good to handle transient errors, but it's better to fail early on non-transient errors (E.g. misconfiguration) and this looks like non-transient. If you really want it to be handled nicely, you would need to retry "a try to parse" at any user request if we could not do it on loading and fail user request if it can't be parsed. But I don't think it's worth the hassle

Reply:
I agree, it is not worth it to do it on every request. And misconfiguration can be caught very early and get fixed.